turbopack/crates/turbopack-core/src/issue/mod.rs

```diff
@@ -147,38 +148,44 @@ pub trait Issue {
     fn source(self: Vc<Self>) -> Vc<OptionIssueSource> {
         Vc::cell(None)
     }
+}
 
-    async fn into_plain(
-        self: Vc<Self>,
-        processing_path: Vc<OptionIssueProcessingPathItems>,
-    ) -> Result<Vc<PlainIssue>> {
-        let description = match *self.description().await? {
-            Some(description) => Some((*description.await?).clone()),
-            None => None,
-        };
-        let detail = match *self.detail().await? {
-            Some(detail) => Some((*detail.await?).clone()),
-            None => None,
-        };
+// A collectible trait that allows traces to be computed for a given module.
+#[turbo_tasks::value_trait]
+pub trait ImportTracer {
+    fn get_traces(self: Vc<Self>, path: ResolvedVc<FileSystemPath>) -> Vc<ImportTraces>;
+}
 
-        Ok(PlainIssue {
-            severity: *self.severity().await?,
-            file_path: self.file_path().to_string().owned().await?,
-            stage: self.stage().owned().await?,
-            title: self.title().owned().await?,
-            description,
-            detail,
-            documentation_link: self.documentation_link().owned().await?,
-            source: {
-                if let Some(s) = &*self.source().await? {
-                    Some(s.into_plain().await?)
-                } else {
-                    None
-                }
-            },
-            processing_path: processing_path.into_plain().await?,
-        }
-        .cell())
+#[turbo_tasks::value(shared)]
+#[derive(Debug)]
+pub struct DelegatingImportTracer {
+    delegates: AutoSet<ResolvedVc<Box<dyn ImportTracer>>>,
+}
+
+impl DelegatingImportTracer {
+    async fn get_traces(&self, path: Vc<FileSystemPath>) -> Result<Vec<ImportTrace>> {
+        Ok(self
+            .delegates
+            .iter()
+            .map(|d| async move { d.get_traces(path).await })
```

Comment:
Nit: Async closures in rust finally got stablized!

```suggestion
            .map(async move |d| d.get_traces(path).await)
```

Reply:
this got rejected because we capture `path`

```
 1  error: async closure does not implement `FnMut` because it captures state from its environment
    --> turbopack/crates/turbopack-core/src/issue/mod.rs:170:18
     |
 170 |             .map(async move |d| d.get_traces(path).await)
     |              --- ^^^^^^^^^^^^^^
     |              |
     |              required by a bound introduced by this call
     |
 note: required by a bound in `std::iter::Iterator::map`
    --> /Users/lukesandberg/.rustup/toolchains/nightly-2025-06-04-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:775:12
     |
 772 |     fn map<B, F>(self, f: F) -> Map<Self, F>
     |        --- required by a bound in this associated function
 ...
 775 |         F: FnMut(Self::Item) -> B,
     |            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::map`

```

looks like this was maybe fixed by https://github.com/rust-lang/rust/pull/123660 ?  im not sure when that will be released